src/full/Agda/TypeChecking/Monad/Base.hs

```diff
@@ -3893,6 +3967,41 @@ aModeToDef AbstractMode = Just AbstractDef
 aModeToDef ConcreteMode = Just ConcreteDef
 aModeToDef _ = Nothing
 
+---------------------------------------------------------------------------
+-- ** Opaque blocks
+---------------------------------------------------------------------------
+
+-- | A block of opaque definitions.
+data OpaqueBlock = OpaqueBlock
+  { opaqueId        :: {-# UNPACK #-} !OpaqueId
+    -- ^ Unique identifier for this block.
+  , opaqueUnfolding :: HashSet QName
+    -- ^ Set of names we are allowed to unfold. After scope-checking,
+    -- this set should be transitively closed.
+  , opaqueDecls     :: HashSet QName
+    -- ^ Declarations contained in this abstract block.
+  , opaqueParent    :: Maybe OpaqueId
+    -- ^ Pointer to an enclosing opaque block, if one exists.
+  , opaqueRange     :: Range
+    -- ^ Where is this opaque block?
+  } deriving (Show, Generic)
+
+instance Pretty OpaqueBlock where
+  pretty (OpaqueBlock _ uf ds p _) = vcat
+    $ [ "opaque (extends " <> pretty p <> ") {"
+      , nest 2 "unfolds"
+      ]
+    ++ [ nest 4 (pretty n <> ",") | n <- HashSet.toList uf ]
```

Comment:
The order given by `HashSet.toList` is dependent on how Agda is built, confer issue
- #6750

Reply:
@plt-amy : I suppose the instability of `HashSet.toList` could be mitigated by piping it through `sort`.  It's just printing, so performance isn't crucial here, stability is more important.

Reply:
Agreed, I'll write a patch.

Reply:
No, I didn't...